controllers/apps/transformer_rbac.go

```diff
@@ -85,39 +85,35 @@ func (c *RBACTransformer) Transform(ctx graph.TransformContext, dag *graph.DAG)
 	for _, compSpec := range componentSpecs {
 		serviceAccountName := compSpec.ServiceAccountName
 		if serviceAccountName == "" {
-			if !needToCreateRBAC(clusterDef) {
+			if !(isProbesEnabled(clusterDef) || isVolumeProtectionEnabled(clusterDef)) {
```

Comment:
put this check out of the loop, it is not the component's checks

Reply:
The procession of serviceAccountName is at the component level, it cannot be moved outside of it.